inference-engine/src/vpu/graph_transformer/src/backend/serialize.cpp

```diff
@@ -63,9 +63,6 @@ int BackEnd::serializeIOInfoSection(
             VPU_INTERNAL_CHECK(data->producerEdge() == nullptr,
                 "serializeIOInfoSection failed on input data {}. Input must have no producer but actually it has: {} with type {}",
                 data->name(), data->producerEdge()->producer()->name(), data->producerEdge()->producer()->type());
-            VPU_INTERNAL_CHECK(data->numConsumers() != 0,
```

Comment:
I find this assertion useful for debugging.
Can we just extend it to check either there are some consumers or count of stages is zero

Reply:
The problem is that input data may have no consumers even if there are some stages in the network. Example: `Parameter->ShapeOf->Result` which is constant folded to `Parameter, Const->Result`, there will be at least one copy between `Const` and `Output`. The problem is that at this moment we can't really tell is it expected for this input to have no consumers, or not, it looks like a separate task of improving the validation of each vpu pass.